internal/controller/isbservicerollout_controller.go

```diff
@@ -208,13 +208,38 @@ func (r *ISBServiceRolloutReconciler) reconcile(ctx context.Context, isbServiceR
 		return err
 	}
 
+	if err = r.applyPodDisruptionBudget(ctx, isbServiceRollout); err != nil {
+		isbServiceRollout.Status.MarkFailed("ApplyISBServiceFailure", err.Error())
+		return fmt.Errorf("failed to apply PodDisruptionBudget for ISBServiceRollout %s, err: %v", isbServiceRollout.Name, err)
+	}
+
 	processISBServiceStatus(ctx, isbsvc, isbServiceRollout)
 
 	isbServiceRollout.Status.MarkRunning()
 
 	return nil
 }
 
+// Apply pod disruption budget for the ISBService
+func (r *ISBServiceRolloutReconciler) applyPodDisruptionBudget(ctx context.Context, isbServiceRollout *apiv1.ISBServiceRollout) error {
+	pdb := kubernetes.NewPodDisruptionBudget(isbServiceRollout.Name, isbServiceRollout.Namespace, 1,
+		[]metav1.OwnerReference{*metav1.NewControllerRef(isbServiceRollout.GetObjectMeta(), apiv1.ISBServiceRolloutGroupVersionKind)},
+	)
+
+	// Create the pdb only if it doesn't exist
+	if err := r.client.Get(ctx, client.ObjectKey{Name: pdb.Name, Namespace: pdb.Namespace}, pdb); err != nil {
```

Comment:
Do we need to update anything if PodDisruptionBudget already found in cluster with given name?

Reply:
update it if the spec is not as expected.

Reply:
Okay